daemon/cmd/endpoint.go

```diff
@@ -561,7 +577,7 @@ func putEndpointIDHandler(d *Daemon, params PutEndpointIDParams) (resp middlewar
 
 	ep.Logger(daemonSubsys).Info("Successful endpoint creation")
 
-	return NewPutEndpointIDCreated()
+	return NewPutEndpointIDOK().WithPayload(fmt.Sprintf("%d", ep.ID))
```

Comment:
I don't understand why we would need to change the API response code in order to implement this feature. `200` is also less accurate than the current `201` we are using, which typically indicates not only that the request was accepted, but that resources were created based on the request.

Reply:
Presumably the existing return code could just be used with the new payload?

Also, for forwards compatibility, the result should probably be a structure rather than a string.

Reply:
> I don't understand why we would need to change the API response code in order to implement this feature.

Because I stored the MAC address in `endpoint`, I need to query `endpoint` to know the MAC address. However, the method `EndpointGet` for querying `endpoint` requires knowing the `endpointID`, and currently there is no way to know the `endpointID`. Therefore, I modified the `EndpointCreate` method to return the ID of newly created `Endpoint`.

Reply:
> Presumably the existing return code could just be used with the new payload?

I didn't quite understand what you said, could you describe it more clearly? Thank you very much.